src/PIL/TiffImagePlugin.py

```diff
@@ -1433,8 +1433,12 @@ def _setup(self) -> None:
         logger.debug("- YCbCr subsampling: %s", self.tag_v2.get(YCBCRSUBSAMPLING))
 
         # size
-        xsize = self.tag_v2.get(IMAGEWIDTH)
-        ysize = self.tag_v2.get(IMAGELENGTH)
+        try:
+            xsize = self.tag_v2[IMAGEWIDTH]
+            ysize = self.tag_v2[IMAGELENGTH]
+        except KeyError as e:
+            msg = "Missing dimensions"
+            raise TypeError(msg) from e
```

Comment:
This is also a new error type that was not raised in past versions, isn't it? So i will also make trouble in Django...

Reply:
Maybe it's a good idea to always raise a special error for all parsers, in all cases where the chunk data will not be enough to get the meta data and use `RuntimeError` as a base, because it's already catch?

Reply:
In 10.4.0, a TypeError would have been raised here by `int(self.tag_v2.get(IMAGEWIDTH))` being run when there was no IMAGEWIDTH tag - effectively, `int(None)`.

This is raising a TypeError again.

It is then [caught in `Image.open` and turned into an `UnidentifiedImageError`](https://github.com/python-pillow/Pillow/blob/82199efbf77f369ad9b7c2c3759e08f517d9925b/src/PIL/Image.py#L3497-L3506), which [inherits from `OSError`](https://github.com/python-pillow/Pillow/blob/82199efbf77f369ad9b7c2c3759e08f517d9925b/src/PIL/__init__.py#L77), and is then [caught in `Parser.feed()`](https://github.com/python-pillow/Pillow/blob/7fe56d62c6739a3aebcc80e2a2cf2c8700fa3726/src/PIL/ImageFile.py#L469-L473), and the Django loop continues.